src/traits/commitment.rs

```diff
@@ -51,8 +51,22 @@ pub trait CommitmentEngineTrait<E: Engine>: Clone + Send + Sync {
   type Commitment: CommitmentTrait<E>;
 
   /// Samples a new commitment key of a specified size
-  fn setup(label: &'static [u8], n: usize) -> Self::CommitmentKey;
+  fn setup(label: &'static [u8], blinding_label: &'static [u8], n: usize) -> Self::CommitmentKey;
 
   /// Commits to the provided vector using the provided generators
   fn commit(ck: &Self::CommitmentKey, v: &[E::Scalar]) -> Self::Commitment;
+
+  /// Commits to the provided vector using the provided generators and random blind
+  fn commit_with_blinding(
```

Comment:
If we make "blinding" the default, we can just update the commit method to accept a blind as an additional input, which will avoid this variant of commit method.

Reply:
That seems good. `src/spartan/snark` and `src/spartan/ppsnark` both need to call an unblinded commit. I can just have then call the new commit method, with 0 as the blind, if that sounds alright?

Reply:
Yes, exactly, this sounds perfect!